src/react-apps/applications/service-development/src/features/deploy/components/deployPaper.tsx

```diff
@@ -0,0 +1,466 @@
+import AltinnIcon from '../../../../../shared/src/components/AltinnIcon';
+import AltinnSpinner from '../../../../../shared/src/components/AltinnSpinner';
+import altinnTheme from '../../../../../shared/src/theme/altinnStudioTheme';
+import { getLanguageFromKey } from '../../../../../shared/src/utils/language';
+import VersionControlContainer from '../../../../../shared/src/version-control/versionControlHeader';
+import { inSyncStatus } from '../containers/deployToTestContainer';
+
+const theme = createMuiTheme(altinnTheme);
+
+const styles = () => createStyles({
+  paperStyling: {
+    padding: 24,
+    maxWidth: 800,
+  },
+  checkIconPositionFix: {
+    position: 'relative',
+    top: '-5px',
+  },
+  bodyTextStyling: {
+    marginLeft: 5,
+    marginTop: 5,
+  },
+  deployButtonInfoText: {
+    color: theme.altinnPalette.primary.grey,
+  },
+  fontSizeTitle: {
+    fontSize: 20,
+  },
+  link: {
+    borderBottom: '1px solid #0062ba',
+  },
+  listItemTitle: {
+    fontSize: theme.overrides.MuiTypography.body1.fontSize,
+    fontWeight: 500,
+  },
+  paperStyleDeployFailed: {
+    backgroundColor: theme.altinnPalette.primary.redLight,
+  },
+  paperStyleDeploySuccess: {
+    backgroundColor: theme.altinnPalette.primary.greenLight,
+  },
+  paperStyleRepoInSync: {
+    backgroundColor: theme.altinnPalette.primary.greyLight,
+  },
+});
+
+interface IDeployPaperProps {
+  classes: any;
+  cSharpCompileStatusSuccess: boolean;
+  cSharpCompileStatusUniqueFilenames: [];
+  deploymentListFetchStatus: any;
+  deployStatus: any;
+  deploySuccess?: boolean;
+  env: string;
+  language: any;
+  localRepoInSyncWithMaster: inSyncStatus.ahead | inSyncStatus.behind | inSyncStatus.ready;
+  masterRepoAndDeployInSync: boolean;
+  onClickStartDeployment: any;
+  titleTypographyVariant: TypographyProps['variant'];
+}
+
+export const DeployPaper = (props: IDeployPaperProps) => {
+  const { classes, localRepoInSyncWithMaster } = props;
+
+  const renderRepoInSync = () => {
+    return (
+      <React.Fragment>
+        <Grid item={true} xs={1} id='renderInSync'>
+          <div
+            className={classNames({
+              [classes.checkIconPositionFix]: localRepoInSyncWithMaster === inSyncStatus.ready,
+            })}
+          >
+            <AltinnIcon
+              iconClass={classNames({
+                ['ai ai-check']: localRepoInSyncWithMaster === inSyncStatus.ready,
+                ['fa fa-info-circle']: localRepoInSyncWithMaster !== inSyncStatus.ready,
+              })}
+              iconColor={localRepoInSyncWithMaster === inSyncStatus.ready ?
+                theme.altinnPalette.primary.green : '#008FD6'}
+              padding='0px 0px 7px 0px'
+            />
+          </div>
+
+        </Grid>
+        <Grid item={true} xs={11}>
+          {renderInSyncText(localRepoInSyncWithMaster)}
+        </Grid>
+      </React.Fragment>
+    );
+  };
+
+  const renderInSyncText = (param: string) => {
```

Comment:
What kind of value does this variable param have? I feel the naming could be more descriptive.

Reply:
Refactored.